demo/src/scala/tuiexamples/UserInputExample.scala

```diff
@@ -99,44 +94,42 @@ object UserInputExample {
           Style.DEFAULT
         )
     }
-    val text = msg.overwrittenStyle(style)
-
-    val help_message = ParagraphWidget(text = text)
-    f.renderWidget(help_message, chunks(0))
-
-    val input = ParagraphWidget(
-      text = Text.nostyle(app.input),
-      block = Some(BlockWidget(borders = Borders.ALL, title = Some(Spans.nostyle("Input")))),
-      style = app.input_mode match {
-        case InputMode.Normal  => Style.DEFAULT
-        case InputMode.Editing => Style.DEFAULT.fg(Color.Yellow)
-      }
-    )
-    f.renderWidget(input, chunks(1))
-
-    app.input_mode match {
-      case InputMode.Normal =>
-        // Hide the cursor. `Frame` does this by default, so we don't need to do anything here
-        ()
-
-      case InputMode.Editing =>
-        // Make the cursor visible and ask tui-rs to put it at the specified coordinates after rendering
-        f.setCursor(
-          // Put cursor past the end of the input text
-          x = chunks(1).x + Grapheme(app.input).width + 1,
-          // Move one line down, from the border to the input line
-          y = chunks(1).y + 1
-        )
-    }
-
     val items: Array[ListWidget.Item] =
       app.messages.zipWithIndex.map { case (m, i) => ListWidget.Item(content = Text.nostyle(s"$i: $m")) }
 
-    val messages =
-      ListWidget(
-        items = items,
-        block = Some(BlockWidget(borders = Borders.ALL, title = Some(Spans.nostyle("Messages"))))
+    Layout
+      .detailed(direction = Direction.Vertical, margin = Margin(2))(
+        Constraint.Length(1) -> ParagraphWidget(text = msg.overwrittenStyle(style)),
+        Constraint.Length(3) -> Widget { (area, buf) =>
```

Comment:
Huh, so I was originally confused about this until I just hit on it myself. For some reason I thought I could put the `Layout` together first and then access the `widgets` and get the cursor positions based off that, but yea that doesn't work obviously. So I'm assuming the main reason you actually do the `Widget` here is so that you can get access to the `area` for the `x` and `y` for the cursor? If that's the only reason then it's a bit of a bummer that we have to do it this way. Although I think maybe this is just something we can deal with now and fix later.

Reply:
O yea, one more thing. One reason actually to not have to do this is given the situation where you maybe have 2 or three different widgets that take input at various times that are part of the same layout. You then have to still wrap them in a `Widget` just for the situation where you _might_ be entering input into that widget at that render. It's hard to explain what I mean, but I can point this out when I finish my migration in skan. I'll shoot up a draft pr and point out a few things.

Sorry, you're just getting comments as I come across them 😬